src/frontend/src/optimizer/plan_node/stream_table_scan.rs

```diff
@@ -189,6 +189,30 @@ impl StreamTableScan {
             .build(vec![0], 1)
             .with_id(state.gen_table_id_wrapped())
     }
+
+    /// Build catalog for cdc backfill state
+    /// Right now we only persist whether the backfill is finished and the corresponding cdc offset
+    /// schema: | `table_id` | `backfill_finished` | `cdc_offset` |
+    pub fn build_cdc_backfill_state_catalog(
+        &self,
+        state: &mut BuildFragmentGraphState,
+    ) -> TableCatalog {
+        let properties = self.ctx().with_options().internal_table_subset();
+        let mut catalog_builder = TableCatalogBuilder::new(properties);
+
+        // use `table_id` as primary key in state table.
+        catalog_builder.add_column(&Field::with_name(DataType::Int64, "table_id"));
```

Comment:
Perhaps this should be associated with `TableId::RW_TYPE`. Similar to how we do for vnodes: `VirtualNode::RW_TYPE`.

Reply:
Considering we may support parallel cdc backfill, perhaps we should generalize the `table_id` column to something like `split_id`.